app/models/assessment.rb

```diff
@@ -443,14 +447,30 @@ def serialize_general
     Utilities.serializable attributes, GENERAL_SERIALIZABLE
   end
 
+  def serialize_time_data
+    s = {}
+    s["start_at"] = self.start_at.strftime('%s')
+    s["visible_at"] = self.visible_at.strftime('%s')
+    s["due_at"] = self.due_at.strftime('%s')
+    s["end_at"] = self.end_at.strftime('%s')
+    s["grading_deadline"] = self.grading_deadline.strftime('%s')
+    s
+  end
+
   def deserialize(s)
-    self.due_at = self.end_at = self.visible_at = self.start_at = self.grading_deadline = Time.now
+    self.start_at = Time.at(s["handin"]["visible_at"].to_i)
+    self.visible_at = Time.at(s["handin"]["start_at"].to_i)
+    self.due_at = Time.at(s["handin"]["due_at"].to_i)
+    self.end_at = Time.at(s["handin"]["end_at"].to_i)
+    self.grading_deadline = Time.at(s["handin"]["grading_deadline"].to_i)
```

Comment:
If you use my suggestion of ISO 8601 timestamps, you can use DateTime.iso8601() here.

Reply:
fair ill swap it out and see if things are stable